Mutagen.Bethesda.Analyzers.Engine/Config/ConfigReader.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Microsoft.Extensions.Logging;
+using Noggog;
+
+namespace Mutagen.Bethesda.Analyzers.Config;
+
+public class ConfigReader<TConfig> : IConfigReader<TConfig> where TConfig : class
+{
+    public const string SettingEqualString = " = ";
+
+    private readonly ILogger<ConfigReader<TConfig>> _logger;
+    private readonly List<Func<TConfig, IReadOnlyList<string>, string, bool>> _processors = [];
+
+    public ConfigReader(ILogger<ConfigReader<TConfig>> logger)
+    {
+        _logger = logger;
+    }
+
+    public ConfigReader<TConfig> Register(Func<TConfig, IReadOnlyList<string>, string, bool> process)
```

Comment:
I think it'll be more functional if this just takes in a `Func<TConfig, IReadOnlyList<string>, string, bool>[]` in the ctor, rather than being mutable.

The idea being when the ConfigReader is made, it is being passed all the processors that will make it up right then and there, rather than always being open to random mutation at some point in the future.

This might be done either via the factory idea, or the labelling of `Func<TConfig, IReadOnlyList<string>, string, bool>` as a `IConfigReaderProcessorUnit<TConfig>` interface or whatever so that ConfigReader can inject `IConfigReaderProcessorUnit[]` as a dependency and get the list of subprocessors upfront that way

Reply:
> The idea being when the ConfigReader is made, it is being passed all the processors that will make it up right then and there, rather than always being open to random mutation at some point in the future.
My plan to go sort of have a Builder pattern for this as it creates a new copy of the ConfigReader every time you register new functions to prevent mutations of any objects that you have. But yeah, also not ideal.